gradle/verification-metadata.xml

```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<verification-metadata xmlns="https://schema.gradle.org/dependency-verification" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.1.xsd">
+   <configuration>
+      <verify-metadata>true</verify-metadata>
+      <verify-signatures>true</verify-signatures>
+      <trusted-artifacts>
+         <trust file="commons-codec-1.4(-sources)*[.](jar|pom)" regex="true"/>
+         <trust file="commons-logging-1.1.1(-sources)*[.](jar|pom)" regex="true"/>
+         <trust file="commons-parent-[1-9]*[.]pom" regex="true"/>
+         <trust file="kxml2-2.3.0-sources.jar" regex="false"/>
+         <trust file="unbescape-1[.]1[.]6[.].*" regex="true"/>
+         <trust file="mockito-core-1.9.5(-sources)*[.](jar|pom)" regex="true"/>
+         <trust file="apache-4.pom" regex="false"/>
+      </trusted-artifacts>
+      <trusted-keys>
+         <trusted-key id="0785b3eff60b1b1bea94e0bb7c25280eae63ebe5" group="org.apache.httpcomponents"/>
+         <trusted-key id="08f0aab4d0c1a4bdde340765b341ddb020fcb6ab" group="org.bouncycastle"/>
+         <trusted-key id="4548945a7793445b67f8f8723b17bfc1fcc1b644" group="io.pebbletemplates"/>
+         <trusted-key id="45dad8aed069092bc7be9843eca081a9790d4fc4" group="org.eclipse.jdt.core.compiler" name="ecj" version="4.2.2"/>
+         <trusted-key id="475f3b8e59e6e63aa78067482c7b12f2a511e325" group="org.slf4j"/>
+         <trusted-key id="4db1a49729b053caf015cee9a6adfc93ef34893e" group="org.hamcrest"/>
+         <trusted-key id="6658dd7f76fcdf7812114f85dc032628e927d006" group="com.android.tools.external.lombok" name="lombok-ast" version="0.2.2"/>
+         <trusted-key id="694621a7227d8d5289699830abe9f3126bb741c1" group="com.google.guava"/>
+         <trusted-key id="90ee19787a7bcf6fd37a1e9180c08b1c29100955" group="com.squareup" name="javawriter" version="2.2.1"/>
+         <trusted-key id="9d23533896a9784703585b6286e02c5a42196ca8" group="org.apache.commons" name="commons-compress" version="1.0"/>
+         <trusted-key id="a5bd02b93e7a40482eb1d66a5f69ad087600b22c" group="org.ow2.asm"/>
+         <trusted-key id="cacfbd4755a2fc78709bdd92be096e29edb8d141" group="net.sf.proguard"/>
+         <trusted-key id="d4c89ea4aaf455fd88b22087efe8086f9e93774e" group="junit" name="junit" version="4.12"/>
+         <trusted-key id="d9e6e4cc8f66b2034995c7192189ca6247f3dee5" group="^com[.]android[.]tools($|([.].*))" regex="true"/>
+         <trusted-key id="e85aed155021af8a6c6b7a4a7c7d8456294423ba" group="org.objenesis" name="objenesis" version="1.0"/>
+         <trusted-key id="fa7929f83ad44c4590f6cc6815c71c0a4e0b8edd" group="net.java.dev.jna"/>
```

Comment:
Can you comment out these trusted keys for the time being and use sha256 hashes for all artifacts?

The sha256 hashes match the hashes in dependencies.properties, so this makes it possible to ensure that what is distributed by auto-update and what’s used during build is the same.

Reply:
Thank you Arne... Actually I didn't manage to get rid of the trusted keys. If I comment them out I get the following error:
Artifact was signed with key 0785b3eff60b1b1bea94e0bb7c25280eae63ebe5 (Oleg Kalnichevski <oleg@ural.ru>, Oleg Kalnichevski <olegk@apache.org>) but this key is not in your trusted key list

Reply:
Thank you for the try! I just checked whether changing the hash of the jna jar triggers a build error and it does, so this suffices.

We need to validate strictly against sha256 for all the shipped dependencies — all which are in dependencies.properties — to ensure that there’s no deviation between our development environments and what nodes actually run.